ovs/extensions/db/arakoon/ArakoonInstaller.py

```diff
@@ -719,3 +722,64 @@ def restart_cluster_remove(cluster_name, remaining_ips):
             if len(remaining_ips) > 2:  # A two node cluster needs all nodes running
                 ArakoonInstaller.wait_for_cluster(cluster_name, client)
         logger.debug('Restart sequence (remove) for {0} completed'.format(cluster_name))
+
+    @staticmethod
+    def is_internal(cluster_type):
+        """
+        Checks if an arakoon cluster is internally managed, if no cluster are found, internally managed is assumed
+        Any cluster found that is externally managed or no result found -> return False
+        :param cluster_type: type of cluster to claim: ['FWK','SD','ABM','NSM']
+        :return: True|False
+        """
+
+        metadata_in_use = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=cluster_type,
+                                                                                in_use=True)
+        metadata_not_in_use = ArakoonInstaller.get_arakoon_metadata_by_cluster_type(cluster_type=cluster_type,
+                                                                                    in_use=False)
+        metadata_in_use.extend(metadata_not_in_use)
+        if metadata_in_use:
+            result = True
+            for metadata in metadata_in_use:
+                if cluster_type in metadata:
+                    result = result and metadata['internal']
+            return result
+
+        return False
+
+    @staticmethod
+    def claim_cluster(cluster_type, cluster_name=''):
+        """
+        Claim an external cluster and mark it in use
+        :param cluster_type: type of cluster to claim: ['FWK','SD','ABM','NSM']
+        :param cluster_name:
+        :return:
+        """
+
+        if ArakoonInstaller.is_internal(cluster_type):
+            key = ArakoonInstaller.metadata_key.format(cluster_name)
+            if cluster_name == '':
+                raise RuntimeError('Cluster name required when marking internal cluster')
+            if not EtcdConfiguration.exists(key):
+                raise RuntimeError('No cluster metadata found for {0}'.format(cluster_name))
+            values = json.loads(EtcdConfiguration.get(key, raw=True))
+            if values['in_use'] == True:
```

Comment:
you could use ```EtcdConfiguration.get(key)``` that would do the ```json.loads(...)``` etc for you. Even more, you can also use stuff like ```EtcdConfiguration.get('/ovs/foo/bar|some.key')``` which will even traverse the json object and return the value of that key. The set works exactly the same. More documentation and examples are in the EtcdConfiguration class' docstring.